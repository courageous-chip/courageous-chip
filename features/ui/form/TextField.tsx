import { useTheme } from "@react-navigation/native";
import React, { memo } from "react";
import {
  FlexStyle,
  StyleSheet,
  TextInput,
  TextInputProps,
  View,
} from "react-native";

type Props = {
  marginBottom?: FlexStyle["marginBottom"];
  placeholder?: TextInputProps["placeholder"];
};

export const TextField = memo<Props>(function ({ marginBottom, placeholder }) {
  const {
    colors: { border: borderColor, card: backgroundColor, text: color },
  } = useTheme();

  return (
    <View
      style={[styles.container, { backgroundColor, borderColor, marginBottom }]}
    >
      <TextInput
        clearButtonMode="while-editing"
        placeholder={placeholder}
        style={[styles.input, { color }]}
      />
    </View>
  );
});

const styles = StyleSheet.create({
  container: {
    borderWidth: 1,
    justifyContent: "center",
    minHeight: 64,
    paddingHorizontal: 10,
  },
  input: {
    fontSize: 14,
    lineHeight: 17,
  },
});
