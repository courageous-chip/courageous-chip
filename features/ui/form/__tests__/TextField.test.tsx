import React from "react";
import renderer from "react-test-renderer";

import { TextField } from "../TextField";

describe("<TextField />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<TextField marginBottom={5} placeholder="ðŸ’ª Biceps Curl" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
