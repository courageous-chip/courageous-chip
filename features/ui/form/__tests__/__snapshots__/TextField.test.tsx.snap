// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderWidth": 1,
        "justifyContent": "center",
        "minHeight": 64,
        "paddingHorizontal": 10,
      },
      Object {
        "backgroundColor": "rgb(255, 255, 255)",
        "borderColor": "rgb(224, 224, 224)",
        "marginBottom": 5,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    clearButtonMode="while-editing"
    placeholder="ðŸ’ª Biceps Curl"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "fontSize": 14,
          "lineHeight": 17,
        },
        Object {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
